Makefile

```diff
@@ -63,34 +74,37 @@ push:
                 echo "Use Local.mk to set the image variable, rebuild, then push."; \
                 exit 0; \
             fi; \
-        fi; \
-        echo docker push ${image}; \
-        docker push ${image}
+        fi
+	docker push ${image}
 	@echo
 
 deploy:
 	@set -e; \
-        pods_gone=False; \
-        kubectl get -o jsonpath='{.items[0].metadata.name}' pods -l name=${project_name} &> /dev/null || pods_gone=True; \
-        if [[ "$$pods_gone" != "True" ]]; then \
-            echo "KubeDirector pod already exists. Maybe old pod is still terminating?"; \
-            exit 1; \
-        fi; \
+        all_namespaces=`kubectl get ns --no-headers| awk '{print $$1}'`; \
```

Comment:
I assume kubedirector will only be running in one namespace? If so, you don't need to loop through all the namespaces. Even if there can be multiple pods, you can get a list of all of them and their namespaces with a single command:

kubectl get pods -l name=kubedirector --all-namespaces --no-headers | awk '{print $1, $2}'

Reply:
About why we are checking everywhere: it's true that KD will only be deployed into one namespace, but there's no guarantee that the current active namespace where THIS deployment attempt is happening is the same as whatever namespace was used for the previous deployment attempt. Once some of us started to deploy KD into kube-system, this bit us a couple of times when changing over to kube-system after doing a test deployment in some other namespace.

Reply:
Good point about the simpler get format. I think I will leave this PR as a straight copy of the changeset though, and do any improvements in a subsequent PR.

Reply:
Tracking in issue #206.